app/src/main/java/uk/ac/bath/petmatch/AppCompatPreferenceActivity.java

```diff
@@ -0,0 +1,109 @@
+package uk.ac.bath.petmatch;
+
+import android.content.res.Configuration;
+import android.os.Bundle;
+import android.preference.PreferenceActivity;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.Nullable;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatDelegate;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A {@link android.preference.PreferenceActivity} which implements and proxies the necessary calls
+ * to be used with AppCompat.
+ */
+public abstract class AppCompatPreferenceActivity extends PreferenceActivity {
```

Comment:
Always extend BaseActivity, otherwise DB, LoginService won't be available and menu should then also be generated in one place..